pkg/cmd/backuprestore/backupnoconfig.go

```diff
@@ -0,0 +1,246 @@
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+
+	gcron "github.com/go-co-op/gocron/v2"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type backupNoConfig struct {
+	kubeConfig    string
+	snapshotExist bool
+	schedule      string
+	retention     backupv1alpha1.RetentionPolicy
+	backupOptions
+}
+
+func NewBackupNoConfigCommand(errOut io.Writer) *cobra.Command {
+	backupNoConf := &backupNoConfig{
+		snapshotExist: false,
+		backupOptions: backupOptions{errOut: errOut},
+	}
+	cmd := &cobra.Command{
+		Use:   "cluster-backup-no-config",
+		Short: "Backs up a snapshot of etcd database and static pod resources without config",
+		Run: func(cmd *cobra.Command, args []string) {
+			must := func(fn func() error) {
+				if err := fn(); err != nil {
+					if cmd.HasParent() {
+						klog.Fatal(err)
+					}
+					fmt.Fprint(backupNoConf.errOut, err.Error())
+				}
+			}
+
+			must(backupNoConf.Validate)
+			must(backupNoConf.Run)
+		},
+	}
+	backupNoConf.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (b *backupNoConfig) AddFlags(fs *pflag.FlagSet) {
+	b.backupOptions.AddFlags(fs)
+}
+
+func (b *backupNoConfig) Validate() error {
+	return b.Validate()
+}
+
+func (b *backupNoConfig) Run() error {
+	backupsClient, err := b.getBackupClient()
+	if err != nil {
+		return err
+	}
+
+	if err = b.extractBackupSpecs(backupsClient); err != nil {
+		return err
+	}
+
+	var errs []error
+	go func() {
+		err := b.scheduleBackup()
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}()
+	go func() {
+		err := b.scheduleBackupPrune()
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}()
+
+	if len(errs) > 0 {
+		return kerrors.NewAggregate(errs)
+	}
+	return nil
+}
+
+func (b *backupNoConfig) getBackupClient() (backupv1client.BackupsGetter, error) {
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", b.kubeConfig)
+	if err != nil {
+		bErr := fmt.Errorf("error loading kubeconfig: %v", err)
+		klog.Error(bErr)
+		return nil, bErr
+	}
+
+	backupsClient, err := backupv1client.NewForConfig(kubeConfig)
+	if err != nil {
+		bErr := fmt.Errorf("error creating etcd backups client: %v", err)
+		klog.Error(bErr)
+		return nil, bErr
+	}
+
+	return backupsClient, nil
+}
+
+func (b *backupNoConfig) extractBackupSpecs(backupsClient backupv1client.BackupsGetter) error {
+	backups, err := backupsClient.Backups().List(context.Background(), v1.ListOptions{})
+	if err != nil {
+		lErr := fmt.Errorf("could not list backup CRDs, error was: [%v]", err)
+		klog.Error(lErr)
+		return lErr
+	}
+
+	if len(backups.Items) == 0 {
+		lErr := fmt.Errorf("no backup CRDs exist, found [%v]", backups)
+		klog.Error(lErr)
+		return lErr
+	}
+
+	idx := slices.IndexFunc(backups.Items, func(backup backupv1alpha1.Backup) bool {
+		return backup.Name == "default"
+	})
+	if idx == -1 {
+		sErr := fmt.Errorf("could not find default backup CR, found [%v]", backups.Items)
+		klog.Error(sErr)
+		return sErr
```

Comment:
that's not an error, if it's not there the user likely has it removed for a reason. We have two options here:
* the sidecar is patched out by the operator when this CRD is not available
* you keep this cron server running without any task to schedule, but you need to have some reconciliation loop to check whether the default CRD comes back eventually

Reply:
if we go with the first option, then we need to patch the CRD client out of this command entirely and use configuration args to pass the CRD (similar to the backup prune command). Otherwise we're going to have issues with race conditions on static pod rollouts when people create+delete the default CRD, which will send this sidecar crash looping.

Reply:
so if we go with second option, that is we return non-err above, why do we need to a `sync()`, as this is a cmd ? 

iiuc,  it does check CRD upon every invocation, or ?

Reply:
how do you envision the lifecycle of this sidecar then? we need to run it 24/7 to keep the cron scheduler running correctly. So when it never restarts, how do we detect the CRD has changed?